.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,899 @@
+        check_name: "Test Results: ubuntu-18_04_java_no-bit"
+        report_paths: OpenDDS/workspace/Tests_JUnit.xml
+
+  # Windows
+
+  windows-2019_defaults_artifact:
+
+    runs-on: windows-2019
+
+    steps:
+    - name: install openssl-windows & xerces-c
+      run: vcpkg install openssl-windows:x64-windows xerces-c:x64-windows
+    - name: checkout MPC
+      uses: actions/checkout@v2
+      with:
+        repository: DOCGroup/MPC
+        path: MPC
+    - name: checkout ACE_TAO
+      uses: actions/checkout@v2
+      with:
+        repository: DOCGroup/ACE_TAO
+        ref: ACE+TAO-6_5_12
+        path: ACE_TAO
+    - name: checkout OpenDDS
+      uses: actions/checkout@v2
+      with:
+        path: OpenDDS
+        submodules: true
+    - name: tweak DDS_TAOv2.mwc to build enough for tests & examples
+      shell: bash
+      run: |
+        sed -i'.old' -e 's/  dds/  $(ACE_ROOT)\/ACEXML\/common\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(ACE_ROOT)\/ACEXML\/parser\/parser\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/utils\/nslist\/nslist.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/tests\/Hello\r\n  dds/' OpenDDS/DDS_TAOv2.mwc # used by java tests
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/tao\/PI_Server\/PI_Server.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/tao\/Strategies\/Strategies.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/orbsvcs\/orbsvcs\/CosNaming.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/orbsvcs\/orbsvcs\/CosNaming_Serv.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/orbsvcs\/orbsvcs\/CosNaming_Skel.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/orbsvcs\/orbsvcs\/FTORB.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/orbsvcs\/orbsvcs\/PortableGroup.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/orbsvcs\/ImplRepo_Service\/ImplRepo_Service.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        sed -i'.old' -e 's/  dds/  $(TAO_ROOT)\/orbsvcs\/Naming_Service\/Naming_Service.mpc\r\n  dds/' OpenDDS/DDS_TAOv2.mwc
+        echo "-----"
+        cat OpenDDS/DDS_TAOv2.mwc
+    - name: configure OpenDDS
+      shell: cmd
+      run: |
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+        cd OpenDDS
+        configure --no-tests --security --ace=%GITHUB_WORKSPACE%/ACE_TAO/ACE --tao=%GITHUB_WORKSPACE%/ACE_TAO/TAO --mpc=%GITHUB_WORKSPACE%/MPC --xerces3=C:\vcpkg\installed\x64-windows --openssl=C:\vcpkg\installed\x64-windows
+    - name: build OpenDDS
+      shell: cmd
+      run: |
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
```

Comment:
Easier is msvc-dev-cmd, see https://github.com/RemedyIT/taox11/blob/bbf9ec8d864aae68e77d9107a84cbd02f974a054/.github/workflows/windows.yml#L57